util/config/config.go

```diff
@@ -459,14 +464,50 @@ func OpenConfigFile() ([]byte, error) {
 
 	// Remove the UTF-8 Byte Order Mark
 	file = bytes.TrimPrefix(file, []byte("\xef\xbb\xbf"))
-	return file, err
+	return file, nil
 
 }
 
-func WriteConfigFile(data []byte) error {
-	configFile := ConfigFile
-	if configFile == "" {
-		configFile = DefaultConfigFile
+func WriteConfigFile(configuration map[string]interface{}) error {
+	bytes, err := json.MarshalIndent(&configuration, "", "    ")
+	if err != nil {
+		return err
 	}
-	return ioutil.WriteFile(configFile, data, 0666)
+	configFile := GetConfigFile()
+	return ioutil.WriteFile(configFile, bytes, 0666)
+}
+
+func SetBeneficiaryAddr(addr string, allowEmpty bool) error {
+	if allowEmpty {
+		if addr == "" {
+			return errors.New("beneficiary address is empty.")
+		}
+	}
+
+	if addr != "" {
+		_, err := ToScriptHash(addr)
+		if err != nil {
+			return errors.New(fmt.Sprintf("parse BeneficiaryAddr error: %v", err))
+		}
+	}
+
+	file, err := OpenConfigFile()
+	if err != nil {
+		return errors.New("Config file not exists.")
```

Comment:
It's better also to return the err here, e.g. `fmt.Errorf("open config file error: %v", err)` or simply just return err